Sources/Sentry/include/SentryThreadHandle.hpp

```diff
@@ -69,6 +71,8 @@ namespace profiling {
          * the threads in the current process, excluding the current thread (the
          * thread that this function is being called on), and the second element
          * is a handle to the current thread.
+         *
+         * Returns ({}, nullptr) if the current thread could not be retrieved.
          */
         static std::pair<std::vector<std::unique_ptr<ThreadHandle>>, std::unique_ptr<ThreadHandle>>
```

Comment:
`h`: If I'm not mistaken, we have to check for a potential `nullptr` of this method here https://github.com/getsentry/sentry-cocoa/blob/48e8c2e10e5b5e103fd28ed34b54ca7a9a7645ca/Sources/Sentry/SentryBacktrace.cpp#L159-L160 or does `backtrace` handle `nullptr` correctly?

Reply:
Good catch, doesn't look like there's a check for nullptr before this dereference.

Reply:
I'm trying to figure out what is the right course of action if that is null...

The `pair` that is returned by `allThreadsExcludingCurrent` contains in the first spot the list of all threads excluding the current thread, and in the second spot, the current thread. So, `pair.second == nullptr` means we couldn't retrieve a handle for the current thread, due to either `pthread_mach_thread_np` returning `nullptr`, or its return value failing the check in `MACH_PORT_VALID`.

The only place `pair.second`, the "calling thread", is used in `backtrace(...)` is to check if the "target thread" is the same as the current (or "calling") thread (this branch is coded as `UNLIKELY`). And in that case, we set the value of the current stack walk frame pointer to `__builtin_frame_address(0)`. After that, we check if that pointer is valid in the current stack bounds (also coded `UNLIKELY`), and if not, we bail early without continuing up the stack, and the stack will effectively be discarded because we don't collect partial stacks.

So if `pair.second` is `nullptr`, should we pass in the current "target" thread instead, which leads to the stack pointer being set to `__builtin_frame_address(0)`? The other alternative is to not walk any of the threads but that doesn't seem right, as there's still data there to collect.

Reply:
Ah, I just noticed that if `allExcludingCurrent` can't get the current thread handle, the pair it returns has null for the current thread (second) and an _empty_ vector in the first part. So, no backtraces will be enumerated in this case. I think that clears up what to do. Just working on a test for this now.

Reply:
Fixed in https://github.com/getsentry/sentry-cocoa/pull/4076